tests/test_wsi_registration.py

```diff
@@ -219,3 +202,40 @@ def test_match_histograms():
     norm_image_a, norm_image_b = match_histograms(image_a, image_b)
     assert np.all(norm_image_a == expected_output)
     assert np.all(norm_image_b == image_b)
+
+
+def test_register(fixed_image, moving_image, fixed_mask, moving_mask):
+    fixed_img = imread(fixed_image)
+    moving_img = imread(moving_image)
+    fixed_msk = imread(fixed_mask)
+    moving_msk = imread(moving_mask)
+
+    df = DFBRegister()
+    with pytest.raises(
+        ValueError,
+        match=r".*The required shape for fixed and moving images is n x m x 3.*",
+    ):
+        _ = df.register(fixed_img[:, :, 0], moving_img[:, :, 0], fixed_msk, moving_msk)
+
+    with pytest.raises(
+        ValueError, match=r".*The input images are expected to have 3 channels.*"
+    ):
+        _ = df.register(
+            fixed_img[:, :, :1], moving_img[:, :, :1], fixed_msk, moving_msk
+        )
```

Comment:
Usually fail cases would be a seperate unit test

Reply:
https://github.com/TissueImageAnalytics/tiatoolbox/blob/e9f33b468d301ebd6c55c3f86bdc9eb03ef6dbcd/tests/test_wsi_registration.py#L241-L292